### PR TITLE
You can now thank the clown car driver, and new medals for being the best clown car driver

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus
add: New medals for being the best clown car driver
add: You can now thank the clown car driver
code: changes usr to user somewhere because seeing usr triggers my bones
/:cl:

couldnt test this because i barely know how this system works

### CODE REVIEW

**Comment by Wilchenx** [comment]
did you thank the clown car driver?

**Comment by ExcessiveUseOfCobblestone** [comment]
to add medals u need to speak to the medal girl @Shadowlight213

File: code/modules/vehicles/vehicle_actions.dm [hunk 1]
```diff
@@ -175,4 +175,19 @@
...
+	desc = "They're just doing their job."
+	button_icon_state = "car_thanktheclown"
+	var/last_thank_time
+
+/datum/action/vehicle/sealed/Thank/Trigger()
+	if(istype(vehicle_entered_target, /obj/vehicle/sealed/car/clowncar))
+		var/obj/vehicle/sealed/car/clowncar/C = vehicle_entered_target
+		if(world.time >= last_thank_time + 60)
+			var/mob/living/carbon/human/clown = pick(C.return_drivers())
+			owner.say("Thank you for the fun ride [clown.name]!")
```
Comment (ATH1909) [inline code/modules/vehicles/vehicle_actions.dm]: Shouldn't there be a comma between "ride" and the driver's name?

Comment (Qustinnus) [reply code/modules/vehicles/vehicle_actions.dm]: yes there should be

**Comment by MMMiracles** [comment]
I had to get ahold of MSO to get the rod suplex medal added to the hub.

**Comment by Qustinnus** [comment]
@MrStonedOne why do we have to use a dumb byond system for this

**Comment by davethwave** [comment]
So this seems to add 2 different medals, one for getting 30 people in the clown car, and another for getting 100 thanks. Once kidnapped, you can thank the clown every 60 seconds via a button if I've read it right.

**Comment by Qustinnus** [comment]
6 seconds

**Comment by Shadowlight213** [comment]
@ExcessiveUseOfCobblestone I gave mso's account maintainer access to the medals page ages ago.
You should ping him, not me for medals.

File: code/modules/vehicles/cars/clowncar.dm [hunk 2]
```diff
@@ -33,6 +35,13 @@
 	. = ..()
 	playsound(src, pick('sound/vehicles/clowncar_load1.ogg', 'sound/vehicles/clowncar_load2.ogg'), 75)
 
+/obj/vehicle/sealed/car/clowncar/after_add_occupant(mob/M, control_flags)
+	. = ..()
+	if(return_controllers_with_flag(VEHICLE_CONTROL_KIDNAPPED).len >= 30)
+		for(var/i in return_drivers())
+			var/mob/voreman = i
```
Comment (PKPenguin321) [inline code/modules/vehicles/cars/clowncar.dm]: @iamgoofball

**Comment by ExcessiveUseOfCobblestone** [comment]
@Shadowlight213 yeah but I miss u :(

**Comment by MrStonedOne** [comment]
Medals added.

File: code/modules/vehicles/cars/clowncar.dm (hunk 2 above)
Comment (Wilchenx) [reply code/modules/vehicles/cars/clowncar.dm]: My clown car is round and full

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**